xarray/coding/cftimeindex.py

```diff
@@ -561,45 +567,68 @@ def to_datetimeindex(self, unsafe=False):
         ------
         ValueError
             If the CFTimeIndex contains dates that are not possible in the
-            standard calendar or outside the nanosecond-precision range.
+            standard calendar or outside the range representable by the
+            specified ``time_unit``.
 
         Warns
         -----
         RuntimeWarning
-            If converting from a non-standard calendar to a DatetimeIndex.
+            If converting from a non-standard calendar, or a Gregorian
+            calendar with dates prior to the reform (1582-10-15).
 
         Warnings
         --------
-        Note that for non-standard calendars, this will change the calendar
-        type of the index.  In that case the result of this method should be
-        used with caution.
+        Note that for non-proleptic Gregorian calendars, this will change the
+        calendar type of the index. In that case the result of this method
+        should be used with caution.
 
         Examples
         --------
         >>> times = xr.cftime_range("2000", periods=2, calendar="gregorian")
         >>> times
         CFTimeIndex([2000-01-01 00:00:00, 2000-01-02 00:00:00],
                     dtype='object', length=2, calendar='standard', freq=None)
-        >>> times.to_datetimeindex()
-        DatetimeIndex(['2000-01-01', '2000-01-02'], dtype='datetime64[us]', freq=None)
+        >>> times.to_datetimeindex(time_unit="ns")
+        DatetimeIndex(['2000-01-01', '2000-01-02'], dtype='datetime64[ns]', freq=None)
         """
 
         if not self._data.size:
             return pd.DatetimeIndex([])
 
-        # transform to us-resolution is needed for DatetimeIndex
-        nptimes = cftime_to_nptime(self, time_unit="us")
+        if time_unit is None:
+            emit_user_level_warning(
+                "In a future version of xarray to_datetimeindex will default "
+                "to returning a 'us'-resolution DatetimeIndex instead of a "
+                "'ns'-resolution DatetimeIndex. This warning can be silenced "
+                "by explicitly setting the time_unit of the index returned.",
```

Comment:
```suggestion
                "by explicitly passing `time_unit` keyword argument.",
```

I read it as "modify the returned object" rather than "pass a kwarg to silence this warning"

Reply:
Thanks @dcherian—indeed that's better.